marimbabot_vision/src/vision_node.py

```diff
@@ -58,13 +61,14 @@ def callbackImage(data: ROSImage, callback_args):
     except CvBridgeError as e:
       rospy.logerror(e)
 
-    detect_notes(cv_image, pre_processor, model)
+    sentence = detect_notes(cv_image, pre_processor, model)
+    sentence_publisher.publish(sentence)
 
     
-def listener(pre_processor, model):
+def listener(pre_processor, model, sentence_publisher):
```

Comment:
What is the purpose of this function? I think the split is not necessary, and you can integrate it into the other part below.